src/HonzaBotner.Discord.Services/EventHandlers/StaffVerificationEventHandler.cs

```diff
@@ -58,36 +51,27 @@ public async Task<EventHandlerResult> Handle(MessageReactionAddEventArgs eventAr
             if (!isAuthenticated)
             {
                 string verificationLink = _urlProvider.GetAuthLink(user.Id, RolesPool.Auth);
-                await channel.SendMessageAsync(
-                    $"Ahoj, ještě nejsi ověřený!\n" +
-                    $"1) Pro ověření ✅ a přidělení rolí dle UserMap klikni na odkaz: {verificationLink}\n" +
-                    "2) Následně znovu klikni na tlačítko 👑 pro přidání zaměstnaneckých rolí.");
+                builder.Content =
+                    "Ahoj, ještě nejsi ověřený!\n" +
+                    "1) Pro ověření ✅ a přidělení rolí dle UserMap klikni na tlačítko dole\n" +
+                    "2) Následně znovu klikni na tlačítko pro přidání 👑 zaměstnaneckých rolí.";
+                builder.AddComponents(new DiscordLinkButtonComponent(verificationLink, "Ověřit se!", false,
+                    new DiscordComponentEmoji("✅")));
+                await eventArgs.Interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource,
+                    builder);
+
                 return EventHandlerResult.Stop;
             }
 
+            await _roleManager.RevokeRolesPoolAsync(eventArgs.User.Id, RolesPool.Staff);
+
             string link = _urlProvider.GetAuthLink(user.Id, RolesPool.Staff);
-            await channel.SendMessageAsync($"Ahoj, pro získání rolí zaměstnance klikni na: {link}");
+            builder.Content = "Ahoj, klikni pro ověření zaměstnaneckých rolí";
+            builder.AddComponents(new DiscordLinkButtonComponent(link, "Jsem zaměstnanec!", false,
+                new DiscordComponentEmoji("👑")));
+            await eventArgs.Interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);
 
             return EventHandlerResult.Stop;
         }
-
-        public async Task<EventHandlerResult> Handle(MessageReactionRemoveEventArgs eventArgs)
-        {
-            if (!(eventArgs.Message.Id == _config.VerificationMessageId
-                  && eventArgs.Message.ChannelId == _config.VerificationChannelId))
-                return EventHandlerResult.Continue;
-            if (!eventArgs.Emoji.Name.Equals(_config.StaffVerificationEmojiName)) return EventHandlerResult.Continue;
-
-            bool revoked = await _roleManager.RevokeRolesPoolAsync(eventArgs.User.Id, RolesPool.Staff);
-            if (!revoked)
-            {
-                _logger.LogWarning("Ungranting roles for user {Username} (id {Id}) failed", eventArgs.User.Username,
-                    eventArgs.User.Id);
-                DiscordDmChannel channel = await eventArgs.Guild.Members[eventArgs.User.Id].CreateDmChannelAsync();
-                await channel.SendMessageAsync("Staff role se nepodařilo odebrat. Prosím, kontaktujte moderátory.");
-            }
-
-            return EventHandlerResult.Continue;
-        }
```

Comment:
Because we are removing this, we should create a different scenario for buttons, considering only staff auth:

1. if not authenticated, send link-button to staff auth
2. if authenticated, send two buttons: link-button with staff auth to refresh roles according to Usermap, and **button to remove staff roles**.

-> aka add that second button and handler to id "staff-verification-remove" or something.

Reply:
Viz staff verification button update:
Addressed, there are now 3 different messages being sent to staff members asking for verification:
Not authenticated -> "You need to authenticate first" + auth button
Authenticated, no staff roles ->"Click to add staff roles" + staff button
Authenticated, with staff roles -> "Click to update/remove staff roles" + staff button AND remove roles button